src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -1186,15 +1207,23 @@ let rec norm : cfg -> env -> stack -> term -> term =
               let delta_level =
                 if s |> BU.for_some (function UnfoldUntil _ | UnfoldOnly _ | UnfoldFully _ -> true | _ -> false)
                 then [Unfold delta_constant]
-                else [NoDelta] in
-              let cfg' = {cfg with steps = ({ to_fsteps s with in_full_norm_request=true})
+                else if cfg.steps.for_extraction
+                then [Env.Eager_unfolding_only; Env.InliningDelta]
```

Comment:
If no other delta steps are set, and we're in extraction, then at least set it to unfolding inline_for_extraction and unfold definitions